speechbrain/nnet/schedulers.py

```diff
@@ -193,6 +193,71 @@ def __call__(self, current_epoch):
         return self.value_at_epoch[old_index], self.value_at_epoch[index]
 
 
+class LinearWarmupScheduler:
```

Comment:
HI and thanks! Is this resumable ? I see a "current_step" shouldn't the sheduler be saved as well in case of resuming experiment? This can be easily done with hooks (see other scheduler with states). What do you think?

Reply:
I agree, on a side note, StepScheduler also does not have hooks. we should fix that in a separate PR.

Reply:
Hi, this is a very good question. TBH, I am not very familiar with the concept of hooks. But I will take a look at how other schedulers are implemented.

Reply:
I have added the checkpint hooks. Please take a look at it.